src/version.c

```diff
@@ -0,0 +1,288 @@
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_GIT_ORIGIN = %s",
+                            LIBWOLFSSL_VERSION_GIT_ORIGIN);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_BRANCH)
+    /* git rev-parse --abbrev-ref HEAD */
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_GIT_BRANCH = %s",
+                            LIBWOLFSSL_VERSION_GIT_BRANCH);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_HASH)
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_GIT_HASH = %s",
+                            LIBWOLFSSL_VERSION_GIT_HASH);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_SHORT_HASH )
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_GIT_SHORT_HASH = %s",
+                            LIBWOLFSSL_VERSION_GIT_SHORT_HASH);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_GIT_HASH_DATE)
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_GIT_HASH_DATE = %s",
+                            LIBWOLFSSL_VERSION_GIT_HASH_DATE);
+#endif
+
+#endif /* else not HAVE_WC_INTROSPECTION */
+    return 0;
+}
+
+/*
+** All platforms: thread details
+*/
+static int ShowExtendedSystemInfo_thread(void)
+{
+    /* all platforms: stack high water mark check */
+#if defined(SINGLE_THREADED)
+    WOLFSSL_VERSION_PRINTF("SINGLE_THREADED");
+#else
+    WOLFSSL_VERSION_PRINTF("NOT SINGLE_THREADED");
+#endif
+    return 0;
+}
+
+/*
+** All Platforms: platform details
+*/
+static int ShowExtendedSystemInfo_platform(void)
+{
+#if defined(WOLFSSL_ESPIDF)
+    #if defined(CONFIG_IDF_TARGET)
+        WOLFSSL_VERSION_PRINTF("CONFIG_IDF_TARGET = %s",
+                                CONFIG_IDF_TARGET);
+        ShowExtendedSystemInfo_platform_espressif(void);
+    #endif
+#endif
+    return 0;
+}
+
+/*
+*******************************************************************************
+** The public ShowExtendedSystemInfo()
+*******************************************************************************
+*/
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+int ShowExtendedSystemInfo(void)
+{
+    WOLFSSL_VERSION_PRINTF("Extended Version and Platform Information.");
+
+#if defined(LIBWOLFSSL_VERSION_STRING)
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_STRING = %s",
+                            LIBWOLFSSL_VERSION_STRING);
+#endif
+
+#if defined(LIBWOLFSSL_VERSION_HEX)
+    WOLFSSL_VERSION_PRINTF("LIBWOLFSSL_VERSION_HEX = %x",
+                            LIBWOLFSSL_VERSION_HEX);
+#endif
+
+#if defined(WOLFSSL_MULTI_INSTALL_WARNING)
+    /* CMake may have detected undesired multiple installs, so give warning. */
+    WOLFSSL_VERSION_PRINTF("");
+    WOLFSSL_VERSION_PRINTF("WARNING: Multiple wolfSSL installs found.");
+    WOLFSSL_VERSION_PRINTF("Check ESP-IDF and local project [components] directory.");
+    WOLFSSL_VERSION_PRINTF("");
+#endif
+
+    ShowExtendedSystemInfo_git(); /* may be limited during active introspection */
+    ShowExtendedSystemInfo_platform();
+    ShowExtendedSystemInfo_thread();
+    return 0;
+}
+
+#endif /* NO_VERSION_EXTENDED_INFO */
+
+#ifdef __cplusplus
```

Comment:
The `__cplusplus` logic only goes in the .h. It is already there, just remove in .c.